### PR TITLE
Added Governance Page to list the maintainers and dev meetings of various OCaml projects

### PR DESCRIPTION
**Pull request by tmattio**
The page lists the teams that compose the ecosystem (Compiler, Packaging, Infrastructure, Platform) and working groups (e.g. Multicore)

This is very WIP, the list of maintainers has been scraped from GitHub teams on the OCaml organisation. I'm opening the PR early to get some early feedback and facilitate communication to get an accurate list of teams and maintainers.

Governance Page | Team | Working Group
-|-|-
screencapture-localhost-8080-governance-2023-06-23-11_40_04 | screencapture-localhost-8080-governance-compiler-2023-06-23-11_40_34 | screencapture-localhost-8080-governance-wg-eio-2023-06-23-11_40_48

### CODE REVIEW

**Comment by sabine** [comment]
I like this. It makes it so much clearer who is a maintainer.

There's a problem with the Email links, they go to 404 pages on ocaml.org.

Would be good to merge after the placeholder texts have been pruned and the lists of maintainers have been completed. Even without the working groups section, if that requires more work. Incremental progress is good.

**Comment by kayceesrk** [comment]
This is great. Thanks @tmattio.

**Comment by tmattio** [comment]
@sabine do you think you'd have time to take over the UI part of this? I can work with the different teams and working groups to get correct data on the maintainers, communication channels, etc.

From a UI perspective, we can aim to get a minimal version in. The most important information to display for each team and subteam, I think, is:

- The list of maintainers
- The communication channels (mailing lists, GitHub, slack channels, discord, etc.)
- The dev meetings if they are open
- The dev meetings notes
- The main repositories they are responsible for

**Comment by sabine** [comment]
Yes. Bring sufficiently complete enough data here and I will fix this up and get it live.

**Comment by avsm** [comment]
This looks great.  Might we be able to use the GitHub API to map the various team ids from the groups there?  The links to email lists on lists.ocaml.org will not currently work either.

**Comment by tmattio** [comment]
> This looks great. Might we be able to use the GitHub API to map the various team ids from the groups there?

That was our idea too, yes! For this, we'd need a few things:

- Move some of the Platform repositories in the Active stage to the OCaml GH organisation (following the Platform governance). In particular ppxlib, ocamlformat, opam-publish and utop
- Create missing teams for the dev teams that don't have one (e.g. opam-dev)
- Move the project permissions to be managed at the GH team level and not at the repository level

Since the GH teams haven't been used uniformly and some of them might be outdated, we were thinking of first agreeing on what the current state of the world is with the different project maintainers, and then open an issue on `ocaml/infrastructure` to request the migration of the projects and updates on the teams.

Happy to proceed differently if you see a better way to do it though.

File: data/governance.yml
Comment (tmattio) [inline data/governance.yml:460]: @aantron do you want to add a working group for Dream here?
@anmonteiro how about one for Melange?

**Review by sabine — APPROVED** [review]

**Comment by avsm** [comment]
Harmonising the GitHub permissions and ensuring this list stays in sync with reality that way sounds good to me.  Note that there are several sub-organisatiosn involved as well -- `ocaml-opam`, `ocaml-ppx`, and `ocaml-dune`.  I have admin on all of those to setup tokens as needed for the API scraping.

File: data/governance.yml
Comment (anmonteiro) [reply data/governance.yml:460]: I'd be OK adding a working group for Melange.

Perhaps including myself and (at least) @jchavarri?

File: data/governance.yml
Comment (Octachron) [inline data/governance.yml]: For the compiler, I think that the the discord link should be removed, since the channel is not official in any way.
Keeping the link to the `caml-list` is debatable, but it is probably a good idea to keep an email channel.

File: data/governance.yml
Comment (sabine) [reply data/governance.yml:460]: Is there a dev meeting or chat for the Melange working group? @anmonteiro

File: data/governance.yml
Comment (sabine) [reply data/governance.yml]: Done!

File: data/governance.yml
Comment (jchavarri) [reply data/governance.yml:460]: There is the [Reason discord](https://discord.gg/reasonml), where there is a specific `#melange` channel.

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    of the language, the standard library, and the compiler tools.
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: opam
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    description: Opam is a source-based package manager. It supports multiple simultaneous compiler installations, flexible package constraints, and a Git-friendly development workflow.
```

I know formatting opam is very strange and doesn’t really adhere to normal grammar rules!

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    description: The OCaml Language Server Protocol (LSP) implementation
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: OCaml-LSP
```
Unless the official name is formatted as the original shows.

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: odoc
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    description: Base library and tools for PPX rewriters
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: UTop
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: opam-publish
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: OCamlFormat
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    description: Streamlining the release of Dune packages to opam
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: dune-release
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: MDX
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: ocp-indent
```
If these name suggestions can’t be in monospace, then I think they could just be formatted as if they were (in all lowercase). What do you think?

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    description: Indentation tool for OCaml to be used with editors like Emacs and Vim.
```
Should this be “from” or “with” editors?

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: OCamlFind
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml:369]:
```suggestion
    description: The OCaml `findlib` library manager 
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: omp
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
  description: The Editors team develops IDE extensions for OCaml. OCaml currently has support for Vim, Emacs, and VSCode.
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    name: `vim-ocaml`
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    packages ecosystem and, in particular, to review submissions to the opam repository.
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml:406]:
```suggestion
    the `opam-repository` CI.
```
Again, if they can be in monospace. If not, leave it as it is.

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
  description: The infrastructure team is responsible for the maintenance and evolution of the infrastructure
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml:462]:
```suggestion
  name: EIO Working Group
```

File: data/governance.yml
Comment (christinerose) [inline data/governance.yml]:
```suggestion
    date: Biweekly on Mondays
```

**Review by christinerose — COMMENTED** [review]
Minor capitalisation/formatting suggestions & a few grammatical corrections. Some of the. Names would normally be in monospace, so I suggested as much; however, if monospace isn’t possible in its final view, then format them as if they were in monospace. If there was a way to visually differentiate them, like bold or italics, then it would be beneficial (but not necessary)

File: data/governance.yml
Comment (sabine) [reply data/governance.yml]: We have no monospace formatting here.

Comment (christinerose) [reply data/governance.yml]: Then just all lowercase. 🙂 (I updated the suggestion)

**Review by christinerose — APPROVED** [review]

**Review by christinerose — APPROVED** [review]

File: data/governance.yml
Comment (rikusilvola) [inline data/governance.yml]:
```suggestion
```
Remove the old mailing list.

File: data/governance.yml
Comment (rikusilvola) [inline data/governance.yml:150]:
```suggestion
    dev-meeting:
      date: Every Monday
      time: 2:00 PM CET
      link: https://us06web.zoom.us/j/84185730263?pwd=TTYrY2puc0dST0lBMjZ5ZHNTaThoZz09
      notes: https://github.com/ocaml/opam/wiki#dev-meetings
    contacts:
```

File: data/governance.yml
Comment (rikusilvola) [inline data/governance.yml:104]:
```suggestion
    dev-meeting:
      date: Every other Wednesday
      time: Alternating 9:00 AM CET and 5:00 PM CET (see wiki)
      link: https://us06web.zoom.us/j/85096877776?pwd=cWNhU1dHQ1ZNSjZuOUZCQ0h2by9Udz09
      notes: https://github.com/ocaml/dune/wiki
    contacts:
```

**Review by christinerose — APPROVED** [review]

**Comment by tmattio** [comment]
The need for that page presented itself again recently.

As mentioned in the review, it would be better to fetch this information from the GitHub API, but until we have up-to-date teams on the different GitHub organisations that we can use to populate the data, I think the page provides enough value that we can keep it updated manually. Therefore, I'll go ahead and merge it now.

While I've tried to get feedback from the maintainers and I believe the list of maintainers represents the current state of the world, it's possible that there are accuracies. If you notice such inaccuracies, first, apologies in advance, and second, please don't hesitate to reach out to or send a pull request!

As a next step for this page:
- We should try to automate it by using GitHub API (see https://github.com/ocaml/infrastructure/issues/55)
- Now that the page is there, we can start adding the existing working groups. For instance, OCaml Wasm, Windows, or any other currently running working group.
- Let's make the information complete and accurate if it's not already the case. If you notice missing maintainers, missing dev meetings, or anything that needs updating, a PR is welcome.

**Pull request merged**
